### ISSUE TITLE
add  languages based ads filters

### ISSUE THREAD

**Issue by net-warriors** [ragibkl/adblock-dns-server#122]
Hello, good morning, please add a language-based ad filter, for example, use this site for Persian language
https://github.com/farrokhi/adblock-iran    
https://github.com/nimasaj/uBOPa
https://github.com/SlashArash/adblockfa 
-------------------------------------------------------------
Do not block ads on sites such as:
example: https://www.zoomg.ir/movie-tv-show-review/341103-hotel-transylvania-4-transformania-review/
https://www.shahrsakhtafzar.com/fa/buyer-guide/1271-best-laptops-for-buy-with-price
 thanks a lot

**Comment by ragibkl** [comment]
Hi @net-warriors , sorry for late reply.

**https://github.com/farrokhi/adblock-iran**
**https://github.com/nimasaj/uBOPa**
For these sources, seems that they provide filters for browser adblock extensions, and also pihole domains. We cannot use the adblock extensions filter list, but I think the pihole domains will work. I'm not sure how effective it will be, but let's try it. **I'll let you know when it's done.**

**https://github.com/SlashArash/adblockfa**
For this source, seems that they only provide filter for browser adblock extensions. Unfortunately, it will not work with dns adblock.

**Comment by net-warriors** [comment]
There are many sources for language-based advertising , It is better to use them
Can't we use a script to convert it to dns ??
https://github.com/nimasaj/uBOPa
https://github.com/farrokhi/adblock-iran

**Comment by ragibkl** [comment]
> There are many sources for language-based advertising , It is better to use them Can't we use a script to convert it for dns ?? https://github.com/nimasaj/uBOPa https://github.com/farrokhi/adblock-iran

Let me explain a little bit about the difference. **Adblock DNS** is quite limited, where we can only block specific domains or sub-domains. **Adblock browser extension** has more features, where they not only can block specific domains, but they can also modify website content, filter out specific url path `example: /api/ads`, or even hide specific elements on the page.

We can only replicate the domains block list from the adblock extensions you provided, inside our dns. In fact, the pihole domains list that I mentioned does exactly that. Anything other than the pihole domains, I think we cannot use them.

I've already included the pihole domains in our Adblock DNS list here: https://github.com/ragibkl/adblock-dns-server/commit/377ef81556161630eb3c3ef615dca2807487eb66. It's also deployed live. If you find any similar list of ad domains that we can block, we can include them.

Regarding the following websites:
- https://www.zoomg.ir/movie-tv-show-review/341103-hotel-transylvania-4-transformania-review/
- https://www.shahrsakhtafzar.com/fa/buyer-guide/1271-best-laptops-for-buy-with-price

I've checked the website using Firefox dev tools. Seems like all the ads are loaded from same domain as the website itself. Unfortunately, if we block these domains in our Adblock DNS, the websites will stop working for us.:
- `cdn.zoomg.ir`
- `www.shahrsakhtafzar.com`

That being said, please test the adblock dns on other websites as well. Maybe some other ads have been blocked by this change?

BTW, can I know which dns servers you are using now? I don't have any servers in Iran, the closest one to you is probably the servers in France. list: https://blog.bancuh.com/adblock-dns/adblock-dns-quickstart/
Or, are you self-host this server?

**Comment by ragibkl** [comment]
Hi @net-warriors ,

I've applied the change here: https://github.com/ragibkl/adblock-dns-server/commit/377ef81556161630eb3c3ef615dca2807487eb66, please test them.

If you have other sources to suggest for your language, do create a new issue ticket for them.

I'll close this issue now, but feel free to open a new one for other issues.
